File: money.rb
class Money
  
  attr_reader :amount
  def initialize(amount)
    @amount = amount
  end

  def buy_in
    @amount = gets.chomp.to_i
  end

  def lose_money
    @amount = @amount - bet
  end
  
  def lose_money_by_half
    @amount = @amount - bet/2
  end

  def win_money
    @amount = @amount + bet
  end

  def win_blackjack
    @amount = @amount + bet * 3 / 2.0
  end

  def to_s
    "#{@amount}"
  end

  def bet
    5
  end
end

File: cardgame2/money.py
class Money:
    def __init__(self, amount):
        self.amount = amount

    def buy_in(self):
        self.amount = int(input())

    def lose_money(self):
        self.amount = self.amount - self.bet()

    def lose_money_by_half(self):
        self.amount = self.amount - self.bet() // 2

    def win_money(self):
        self.amount = self.amount + self.bet()

    def __str__(self):
        return str(self.amount)

    def bet(self):
        return 5